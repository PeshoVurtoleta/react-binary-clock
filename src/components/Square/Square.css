.Clock__Square {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #FF1C43;
    border-radius: 50%;
}

.Clock__Square--lit {
    background-color: #fbebaf;
    color: #FF1C43;
}