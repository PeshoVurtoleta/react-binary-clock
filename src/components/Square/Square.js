import React from "react";

import './Square.css';

export default function Square({on, column, row, number}) {
    const gridRow = row ? ` grid__row-${row}` : "";
    const lit = on ? " Clock__Square--lit" : "";

    return <span className={`Clock__Square grid__column-${column}${gridRow}${lit}`}>
            {on && number}
        </span>
};