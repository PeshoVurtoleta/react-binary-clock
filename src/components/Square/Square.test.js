import React from 'react';

import Square from './Square';

import {shallow} from 'enzyme';

describe(('Should render correctly:'), () => {
    test('Should render correctly:', () => {
        const component = shallow(<Square />);

        expect(component.exists()).toBeTruthy();
        expect(component).toMatchSnapshot();
    });

    test('Should render correctly with all modifiers:', () => {
        const on = true;
        const row = 3;
        const column = 2;
        const number = 4;

        const component = shallow(<Square {...{on, row, column, number}} />);

        expect(component).toMatchSnapshot();
    });

    test('Should render correctly with no row:', () => {
        const on = true;
        const column = 2;
        const number = 4;

        const component = shallow(<Square {...{on, column, number}} />);

        expect(component).toMatchSnapshot();
    });
});