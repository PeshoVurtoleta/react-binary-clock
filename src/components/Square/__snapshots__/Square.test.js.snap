// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly: Should render correctly with all modifiers: 1`] = `
<span
  className="Clock__Square grid__column-2 grid__row-3 Clock__Square--lit"
>
  4
</span>
`;

exports[`Should render correctly: Should render correctly with no row: 1`] = `
<span
  className="Clock__Square grid__column-2 Clock__Square--lit"
>
  4
</span>
`;

exports[`Should render correctly: Should render correctly: 1`] = `
<span
  className="Clock__Square grid__column-undefined"
/>
`;
